### PR TITLE
astyle validation

### PR DESCRIPTION
**Pull request by jupe**
## Status
**READY**

## Description
Validate in CI that code style is valid and correspond with Mbed OS style rules using [astyle](http://astyle.sourceforge.net).

### CODE REVIEW

**Comment by jupe** [comment]
seems that astyle is not yet in place in this slave
```
+ astyle source/mbed_trace.c mbed-trace/mbed_trace.h --project=.astylerc
/builds/ws/race_PR-89-3DFX3CBTFKFJI5WE5DP56@5/mbed-trace@tmp/durable-8d7aca58/script.sh: 2: /builds/ws/race_PR-89-3DFX3CBTFKFJI5WE5DP56@5/mbed-trace@tmp/durable-8d7aca58/script.sh: astyle: not found
```

File: Jenkinsfile [hunk 1]
```diff
@@ -158,11 +158,10 @@ def yottaBuildStep(target, compilerLabel) {
               execute("gcovr -x -o junit.xml")
               execute("cppcheck --enable=all --std=c99 --inline-suppr --template=\"{file},{line},{severity},{id},{message}\" source 2> cppcheck.txt")
 
-              // Waiting for IOTTESTINF-4864
               // check if astyle is correct
-              // execute("astyle source/*.c mbed-trace/*.h --project=.astylerc")
+              execute("astyle --options=.astylerc source/*.c mbed-trace/*.h")
               // check differency
-              // execute("git diff-index --quiet HEAD")
+              execute("git diff-index --quiet HEAD")
```
Comment (artokin) [inline Jenkinsfile]: Is this output enough for the developer to see what needs to be corrected?

Comment (jupe) [reply Jenkinsfile]: yeah, maybe should not be `--quiet`..

Comment (jupe) [reply Jenkinsfile]: removed `--quiet` . I'll double check that it gives enough information if astyle is not clean..

Comment (jupe) [reply Jenkinsfile]: using now `git diff-index -p --exit-code HEAD`
results as patch format like :

```
[mbed-trace] Running shell script
+ git diff-index -p --exit-code HEAD
diff --git a/source/mbed_trace.c b/source/mbed_trace.c
index 8a19d1b..f90e0ea 100644
--- a/source/mbed_trace.c
+++ b/source/mbed_trace.c
@@ -152,7 +152,7 @@ static trace_t m_trace = {
     .mutex_lock_count = 0
 };
 
-int mbed_trace_init(void )
+int mbed_trace_init(void)
 {
     if (m_trace.line == NULL) {
         m_trace.line = MBED_TRACE_MEM_ALLOC(m_trace.line_length);
```

Better?

Comment (artokin) [reply Jenkinsfile]: much better, thanks

**Pull request merged**